test/extended/apiserver/resiliency.go

```diff
@@ -0,0 +1,132 @@
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/transport"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+
+	"github.com/openshift/origin/test/extended/single_node"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+func forceApiRollout(oc *exutil.CLI) {
+	redeploymentReason := fmt.Sprintf(`{"spec":{"forceRedeploymentReason":"resiliency-test-%v"}}`, uuid.NewUUID())
+
+	_, err := oc.AdminOperatorClient().OperatorV1().KubeAPIServers().Patch(context.Background(), "cluster", types.MergePatchType,
+		[]byte(redeploymentReason), metav1.PatchOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+}
+
+func createHttpClient(clusterConfig *rest.Config, timeout time.Duration) (*http.Client, error) {
+	kubeTransportConfig, err := clusterConfig.TransportConfig()
+	if err != nil {
+		return nil, err
+	}
+	tlsConfig, err := transport.TLSConfigFor(kubeTransportConfig)
+	if err != nil {
+		return nil, err
+	}
+	httpTransport := &http.Transport{
+		Dial: (&net.Dialer{
+			Timeout: timeout,
+		}).Dial,
+		TLSClientConfig:     tlsConfig,
+		TLSHandshakeTimeout: timeout,
+		IdleConnTimeout:     timeout,
+	}
+
+	roundTripper := http.RoundTripper(httpTransport)
+	if kubeTransportConfig.HasTokenAuth() {
+		roundTripper, err = transport.NewBearerAuthWithRefreshRoundTripper(kubeTransportConfig.BearerToken, kubeTransportConfig.BearerTokenFile, httpTransport)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &http.Client{
+		Transport: roundTripper,
+	}, nil
+}
+
+func isApiReady(clusterConfig *rest.Config, httpClient *http.Client) (bool, error) {
+	resp, err := httpClient.Get(clusterConfig.Host + "/readyz")
+	if resp != nil && resp.Body != nil {
+		defer resp.Body.Close()
+	}
+
+	if err != nil {
+		return false, err
+	}
+
+	if resp.StatusCode < 200 || resp.StatusCode >= 400 {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err == nil {
+			err = fmt.Errorf("error running request: %v: %v", resp.Status, string(body))
+		} else {
+			err = fmt.Errorf("error running request: %v", resp.Status)
+		}
+
+		return false, err
+	}
+
+	return true, nil
+}
+
+var _ = ginkgo.Describe("[Conformance][sig-sno][Serial] Cluster", func() {
+	f := framework.NewDefaultFramework("cluster-resiliency")
+	f.SkipNamespaceCreation = true
+	f.SkipPrivilegedPSPBinding = true
+
+	oc := exutil.NewCLIWithoutNamespace("cluster-resiliency")
+
+	ginkgo.It("should allow a fast rollout", func() {
+		controlPlaneTopology, infraTopology := single_node.GetTopologies(f)
+
+		if controlPlaneTopology != configv1.SingleReplicaTopologyMode && infraTopology != configv1.SingleReplicaTopologyMode {
+			e2eskipper.Skipf("Test is only relevant for single replica topologies")
```

Comment:
isn't a check for controlPlaneTopology enough? We are verifying the control plane here, don't we?